my-app/src/AddBox/AddBox.jsx

```diff
@@ -0,0 +1,180 @@
+import React, { Component } from 'react';
+import Autosuggest from 'react-autosuggest';
+
+//import isMobile from 'ismobilejs';
+
+//const focusInputOnSuggestionClick = !isMobile.any;
+const uniqLabelFast = a => {
+  /**
+   * Quickly makes a list tags without repeating labels
+   * @param a {[Tag]} list of tags
+   */
+
+    var seen = {};
+    var out = [];
+    var len = a.length;
+    var j = 0;
+    for(var i = 0; i < len; i++) {
+         var item = a[i].label;
+         if(seen[item] !== 1) {
+               seen[item] = 1;
+               out[j++] = a[i];
+         }
+    }
+    return out;
+}
+
+const escapeRegexCharacters = str => str.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
```

Comment:
What is this escaping trying to protect against?  Shell?  HTML?  Things that can't appear in JSON?  I think there might be a standard library (or dependency we can import) that does the right kind of escaping and is already well tested.